templates/lib/install.sh

```diff
@@ -0,0 +1,51 @@
+set -ex
+
+main() {
+  local target=
+  if [ $TRAVIS_OS_NAME = linux ]; then
+    target=x86_64-unknown-linux-musl
+    sort=sort
+  else
+    target=x86_64-apple-darwin
+    sort=gsort  # for `sort --sort-version`, from brew's coreutils.
+  fi
+
+  # Builds for iOS are done on OSX, but require the specific target to be
+  # installed.
+  case $TARGET in
+    aarch64-apple-ios)
+      rustup target install aarch64-apple-ios
+      ;;
+    armv7-apple-ios)
+      rustup target install armv7-apple-ios
+      ;;
+    armv7s-apple-ios)
+      rustup target install armv7s-apple-ios
+      ;;
+    i386-apple-ios)
+      rustup target install i386-apple-ios
+      ;;
+    x86_64-apple-ios)
+      rustup target install x86_64-apple-ios
+      ;;
+  esac
+
+  # This fetches latest stable release
+  local tag=$(git ls-remote --tags --refs --exit-code https://github.com/japaric/cross \
+    | cut -d/ -f3 \
+    | grep -E '^v[0.1.0-9.]+$' \
+    | $sort --version-sort \
+    | tail -n1)
+  curl -LSfs https://japaric.github.io/trust/install.sh | \
+    sh -s -- \
+    --force \
+    --git japaric/cross \
+    --tag $tag \
+    --target $target
+
+  # Install test dependencies
+  rustup component add rustfmt-preview
+  rustup component add clippy-preview
```

Comment:
What if we also install `bindgen` and setup a build stage responsible to build the `.h` headers and publish on Github?

Reply:
A few quick notes here:
- we would need `cbindgen`, not `bindgen` here (e.g. wrap `rust` code to be consumed from `C`, not the other way around)
- [here's an example of how to do that](https://github.com/Gankro/tutorial-cbindgen)
- in the example it stands out to me that a [`cbindgen.toml`](https://github.com/Gankro/tutorial-cbindgen/blob/master/rust-bindings/cbindgen.toml) file is included. Should we generate that as well, or leave that up to the user?
- building with `cbindgen` can either be done [from the command line](https://github.com/Gankro/tutorial-cbindgen/blob/master/build.sh#L5) or from [build.rs](https://github.com/Gankro/tutorial-cbindgen/blob/master/rust-bindings/optional-build.rs), I'm not sure which one is preferable. But we should probably pick one.